lib/api/controllers/securityController.js

```diff
@@ -819,20 +820,46 @@ class SecurityController extends NativeController {
       ? null
       : request.getBodyArray('profileIds');
 
-    const updated = await this.ask(
-      'core:security:user:update',
-      id,
-      profileIds,
-      content,
-      {
-        refresh: request.getRefresh('wait_for'),
-        retryOnConflict: request.getInteger('retryOnConflict', 10),
-        userId,
-      });
+    return this._changeUser(request, id, content, userId, profileIds);
+  }
 
-    global.kuzzle.log.info(`[SECURITY] User "${userId}" applied action "${request.input.action}" on user "${id}."`);
+  /**
+   * Applies a partial update to an existing user.
+   * If the user doesn't already exist, a new user is created.
+   *
+   * @param {Request} request
+   * @returns {Promise}
+   */
+  async upsertUser (request) {
+    const id = request.getId();
+    const content = request.getBodyObject('content');
+    const userId = request.getKuid();
+    const profileIds = get(content, 'profileIds');
```

Comment:
I just realized that this and l.840 to 846 are leftovers from a code before we had argument extractors in the request (`request.getBodyXXX`)

You should retrieve the `profileIds` array with `request.getBodyArray` method instead